pkg/roles/handlers.go

```diff
@@ -0,0 +1,458 @@
+	mux.Delete("/api/v0/roles/{id}/entitlements/{e_id}", a.handleRemovePermission)
+	mux.Get("/api/v0/roles/{id}/groups", a.handleListRoleGroup)
+
+}
+
+func (a *API) handleList(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	user := r.Context().Value(authorization.USER_CTX).(*authorization.User)
+
+	roles, err := a.service.ListRoles(
+		r.Context(),
+		user.ID,
+	)
+
+	if err != nil {
+		rr := types.Response{
+			Status:  http.StatusInternalServerError,
+			Message: err.Error(),
+		}
+
+		w.WriteHeader(http.StatusInternalServerError)
+		json.NewEncoder(w).Encode(rr)
+
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+
+	json.NewEncoder(w).Encode(
+		types.Response{
+			Data:    roles,
+			Message: "List of roles",
+			Status:  http.StatusOK,
+		},
+	)
+}
+
+func (a *API) handleDetail(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	ID := chi.URLParam(r, "id")
+
+	user := r.Context().Value(authorization.USER_CTX).(*authorization.User)
+
+	role, err := a.service.GetRole(r.Context(), user.ID, ID)
+
+	if err != nil {
+
+		rr := types.Response{
+			Status:  http.StatusInternalServerError,
+			Message: err.Error(),
+		}
+
+		w.WriteHeader(http.StatusInternalServerError)
+		json.NewEncoder(w).Encode(rr)
+
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(
+		types.Response{
+			Data:    []string{role},
+			Message: "Rule detail",
+			Status:  http.StatusOK,
+		},
+	)
+}
+
+func (a *API) handleCreate(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	defer r.Body.Close()
+	body, err := io.ReadAll(r.Body)
+
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		json.NewEncoder(w).Encode(
+			types.Response{
+				Message: "Error parsing request payload",
+				Status:  http.StatusBadRequest,
+			},
+		)
+
+		return
+	}
+
+	role := new(RoleRequest)
+	if err := json.Unmarshal(body, role); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		json.NewEncoder(w).Encode(
+			types.Response{
+				Message: "Error parsing JSON payload",
+				Status:  http.StatusBadRequest,
+			},
+		)
+
+		return
+
+	}
+
+	user := r.Context().Value(authorization.USER_CTX).(*authorization.User)
+
+	err = a.service.CreateRole(r.Context(), user.ID, role.ID)
+
+	if err != nil {
+
+		rr := types.Response{
+			Status:  http.StatusInternalServerError,
+			Message: err.Error(),
+		}
+
+		w.WriteHeader(http.StatusInternalServerError)
+		json.NewEncoder(w).Encode(rr)
+
+		return
+	}
+
+	w.WriteHeader(http.StatusCreated)
+	json.NewEncoder(w).Encode(
+		types.Response{
+			Message: fmt.Sprintf("Created role %s", role.ID),
+			Status:  http.StatusCreated,
+		},
+	)
+}
+
+// handleUpdate is not implemented by choice, product might decide to do it to enhcance
+// role metadata, we do not support anything on top of simple ID attribute and this is
+// not changeable right now due to coupled implemntation with OpenFGA
```

Comment:
s/implemntation/implementation/

Reply:
if this is not implemented, then just omit this method.. or return a not implemented error